src/ZkFold/Symbolic/Compiler/ArithmeticCircuit.hs

```diff
@@ -48,52 +50,50 @@ import           ZkFold.Prelude                                         (length)
 import           ZkFold.Symbolic.Compiler.ArithmeticCircuit.Combinators (desugarRange)
 import           ZkFold.Symbolic.Compiler.ArithmeticCircuit.Instance    ()
 import           ZkFold.Symbolic.Compiler.ArithmeticCircuit.Internal    (Arithmetic, ArithmeticCircuit (..), Constraint,
-                                                                         apply, eval, eval1, exec, exec1,
+                                                                         Var (..), acInput, eval, eval1, exec, exec1,
                                                                          witnessGenerator)
 import           ZkFold.Symbolic.Compiler.ArithmeticCircuit.Map
 
 --------------------------------- High-level functions --------------------------------
 
 -- TODO: make this work for different input types.
-applyArgs :: ArithmeticCircuit a f -> [a] -> ArithmeticCircuit a f
-applyArgs r args =
-    (execState (apply args) r {acOutput = U1}) {acOutput = acOutput r}
+-- applyArgs :: ArithmeticCircuit a (i :*: j) o -> i a -> ArithmeticCircuit a j o
```

Comment:
Turns out that this partial application function was not used anywhere but in tests, but there a complete application (`eval`) was used actually.

Also, for partial application purposes, `Symbolic` functions + `FromConstant` instances work better and more intuitively in my opinion, maybe we use it from here on out?

Reply:
removed commented code